client/store/bikes.js

```diff
@@ -108,9 +121,33 @@ const getOneBike = bike => ({type: GET_SINGLE_BIKE, bike})
  * REDUCER
  */
 export default function(state = initialState, action) {
+  let newFilteredBikes = []
+  let categoryVal
   switch (action.type) {
     case GET_BIKES:
-        return {...state, bikes: action.bikes}
+        return {...state, bikes: action.bikes, filteredBikes: action.bikes}
+    case ADD_FILTER:
+        categoryVal = parseInt(action.categoryVal) // we want this as a number not string
+        if (!state.selectedCategoryVals.includes(categoryVal)) { // add the category val if it doesn't exist
+          return {...state, selectedCategoryVals: [...state.selectedCategoryVals, categoryVal]}
+        } else { // remove categoryVal if it exists
+          return {...state, selectedCategoryVals: [...state.selectedCategoryVals.filter(val => val !== categoryVal)]}
+        }
+    case FILTER_BIKES:
+        // if we have no selectedCategoryVals
+        if (state.selectedCategoryVals.length <= 0) {
+          return {...state, filteredBikes: [...state.bikes]}
+        } else {
+          [...state.bikes].forEach(bike => {
```

Comment:
it doesn't look like this else clause is returning a state, that looks like a problem.  can't see how filtered bikes is finding its way through back to the all bikes view.